app/Actions/Photo/Create.php

```diff
@@ -134,11 +134,7 @@ protected function loadFileMetadata(SourceFileInfo $sourceFileInfo)
 
 		// Use title of file if IPTC title missing
 		if ($this->strategyParameters->info['title'] === '') {
-			if ($this->strategyParameters->kind == 'raw') {
-				$this->strategyParameters->info['title'] = substr(basename($sourceFileInfo->getOriginalFilename()), 0, 98);
-			} else {
-				$this->strategyParameters->info['title'] = substr(basename($sourceFileInfo->getOriginalFilename(), $sourceFileInfo->getOriginalFileExtension()), 0, 98);
-			}
+			$this->strategyParameters->info['title'] = $sourceFileInfo->getOriginalName();
```

Comment:
You are removing a `substr($s, 0 , 98)`
This will create problems if the name is longer than 100 characters.
I did not see any checks in that regard.

Reply:
Nicely spotted. I re-added the `substr` which was lost.